lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -0,0 +1,113 @@
+"""
+Tests for the Tool Data Meister modules.
+"""
+import pathlib
+import tarfile
+import pytest
+
+from pbench.agent.tool_meister import ToolMeister, get_logger
+
+
+@pytest.fixture()
+def get_params():
+    a = {
+        "benchmark_run_dir": "",
+        "channel_prefix": "",
+        "tds_hostname": "test.hostname.com",
+        "tds_port": 4242,
+        "controller": "test.hostname.com",
+        "group": "",
+        "hostname": "test.hostname.com",
+        "label": None,
+        "tool_metadata": {"persistent": {}, "transient": {}},
+        "tools": [],
+    }
+    return a
+
+
+def test_create_tar(agent_setup, get_params):
+    """Test create tar file"""
+    tm = ToolMeister(
+        pbench_install_dir=None,
+        tmp_dir=None,
+        tar_path="/usr/bin/tar",
+        sysinfo_dump=None,
+        params=get_params,
+        redis_server=None,
+        logger=None,
+    )
+    tmp_dir = agent_setup["tmp"]
+
+    # create a file in tmp directory
+    (tmp_dir / "file1").write_text("")
```

Comment:
Assuming that we are mocking out the subroutines that `_create_tar()` calls, we shouldn't actually need a temp directory or any file system artifacts.

On the flip side, I don't see the mock for `subprocess.run()` here....